### ISSUE TITLE
subroutines as second arg for LOGCONFESS and LOGCROAK do not work as expected in easy mode

### ISSUE THREAD

**Issue by JC001010** [polettix/Log-Log4perl-Tiny#12]
Using subroutines as second arg for a log message doesn't work as expected when using any of the LOG<Carp> functions in easy mode. It shows the subroutine address instead of the returned value. Others such as LOGDIE work fine. See example here.

```
use Log::Log4perl::Tiny ":easy";

# This doesn't work:
LOGCONFESS "Names: ", sub{  join(",", qw/John Jane Mario/) };

#This works:
LOGDIE "Names: ", sub{  join(",", qw/John Jane Mario/) };
```

Both fail when writing to STDERR, but LOGDIE works fine when adding the messages to a file.

**Comment by polettix** [comment]
Thanks for reporting, I'll look into this shortly!

**Comment by polettix** [comment]
This should hopefully fix it: https://github.com/polettix/Log-Log4perl-Tiny/tree/2de207c931ac2d984583df33ec99097fde36d669

I've sent a trial release 1.4.1 to CPAN, it should be available shortly. It does not test the easy mode stuff because the problem was at the underlying level, please let me know if there are still residual issues (feel free to contribute a test case! :smile:)

**Comment by polettix** [comment]
Should now be fixed in 1.6.0, please reopen if this is not the case. Thanks again! 😄
